include/ignition/math/Vector4.hh

```diff
@@ -126,6 +128,56 @@ namespace ignition
         this->data[3] = _w;
       }
 
+      /// \brief Set this vector's components to the maximum of itself and the
+      ///        passed in vector
+      /// \param[in] _v the maximum clamping vector
+      public: void Max(const Vector4<T> &_v)
+      {
+        if (_v[0] > this->data[0])
+          this->data[0] = _v[0];
```

Comment:
I believe this would be more concise

```suggestion
        this->data[0] = std::max(_v[0], this->data[0]);
```

Reply:
I thing the same, and prefer `std::min`, only did with the ifs because is implemented this way in Vector3. Despite that I can use `std::min`?

Reply:
I prefer this, it would actually probably make sense to go back and fix vector3 and vector2 to be the same, but it's out of the scope of this pr.